### PR TITLE
Adds consumer

### PR DESCRIPTION
**Pull request by markturansky**
Adds Consumer Kind

```
$ make test && make test-integration
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -v  \
	./pkg/... \
	./cmd/...
EMPTY pkg/api
EMPTY pkg/api/openapi
EMPTY pkg/api/presenters
EMPTY pkg/auth
EMPTY pkg/client/ocm
PASS pkg/config.TestConfigReadStringFile (0.00s)
PASS pkg/config.TestConfigReadIntFile (0.00s)
PASS pkg/config.TestConfigReadBoolFile (0.00s)
PASS pkg/config.TestConfigReadQuotedFile (0.00s)
PASS pkg/config
PASS pkg/controllers.TestControllerFramework (0.00s)
PASS pkg/controllers
EMPTY pkg/dao
EMPTY pkg/dao/mocks
EMPTY pkg/db
EMPTY pkg/db/db_context
EMPTY pkg/db/db_session
EMPTY pkg/db/migrations
EMPTY pkg/db/mocks
EMPTY pkg/db/transaction
PASS pkg/errors.TestErrorFormatting (0.00s)
PASS pkg/errors.TestErrorFind (0.00s)
PASS pkg/errors
EMPTY pkg/handlers
EMPTY pkg/logger
PASS pkg/services.TestDinosaurFindBySpecies (0.00s)
PASS pkg/services.TestSQLTranslation (0.00s)
PASS pkg/services
EMPTY pkg/util
EMPTY cmd/maestro
EMPTY cmd/maestro/clone
PASS cmd/maestro/environments.TestLoadServices (0.27s)
PASS cmd/maestro/environments
EMPTY cmd/maestro/migrate
EMPTY cmd/maestro/servecmd
EMPTY cmd/maestro/server
EMPTY cmd/maestro/server/logging

DONE 10 tests in 6.107s
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
		./test/integration
I1114 10:11:03.672876   57252 framework.go:72] Initializing testing environment
I1114 10:11:03.913949   57252 framework.go:152] Using Mock OCM Authz Client
I1114 10:11:03.913957   57252 framework.go:177] Disabling Sentry error reporting
I1114 10:11:03.917845   57252 api_server.go:145] Serving without TLS at localhost:8000
PASS test/integration.TestConsumerPost (0.24s)
PASS test/integration.TestControllerRacing (1.40s)
PASS test/integration.TestDinosaurGet (0.29s)
PASS test/integration.TestDinosaurPost (0.24s)
PASS test/integration.TestDinosaurPatch (0.25s)
PASS test/integration.TestDinosaurPaging (0.39s)
PASS test/integration.TestDinosaurListSearch (0.29s)
PASS test/integration.TestUpdateDinosaurWithRacingRequests (0.31s)
PASS test/integration.TestUpdateDinosaurWithRacingRequests_WithoutLock (0.39s)
I1114 10:11:07.724129   57252 api_server.go:151] Web server terminated
PASS test/integration

DONE 9 tests in 5.233s
```

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -187,6 +187,182 @@ paths:
...
     parameters:
     - $ref: '#/components/parameters/id'
+  /api/maestro/v1/consumers:
+    get:
+      summary: Returns a list of consumers
+      security:
+        - Bearer: []
+      responses:
+        '200':
+          description: A JSON array of dinosaur objects
```
[the remark this answers is not in the document]
Comment (markturansky) [reply openapi/openapi.yaml]: fixed

File: openapi/openapi.yaml [hunk 2]
```diff
@@ -187,6 +187,182 @@ paths:
...
+                $ref: '#/components/schemas/Error'
+        '500':
+          description: An unexpected error occurred creating the dinosaur
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/maestro/v1/consumers/{id}:
+    get:
+      summary: Get an dinosaur by id
```
[the remark this answers is not in the document]
Comment (markturansky) [reply openapi/openapi.yaml]: fixed

**Comment by markturansky** [comment]
```
$ make test && make test-integration
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -v  \
        ./pkg/... \
        ./cmd/...
EMPTY pkg/api
EMPTY pkg/api/openapi
EMPTY pkg/api/presenters
EMPTY pkg/auth
EMPTY pkg/client/ocm
PASS pkg/config.TestConfigReadStringFile (0.00s)
PASS pkg/config.TestConfigReadIntFile (0.00s)
PASS pkg/config.TestConfigReadBoolFile (0.00s)
PASS pkg/config.TestConfigReadQuotedFile (0.00s)
PASS pkg/config (cached)
PASS pkg/controllers.TestControllerFramework (0.00s)
PASS pkg/controllers (cached)
EMPTY pkg/dao
EMPTY pkg/dao/mocks
EMPTY pkg/db
EMPTY pkg/db/db_context
EMPTY pkg/db/db_session
EMPTY pkg/db/migrations
EMPTY pkg/db/mocks
EMPTY pkg/db/transaction
PASS pkg/errors.TestErrorFormatting (0.00s)
PASS pkg/errors.TestErrorFind (0.00s)
PASS pkg/errors (cached)
EMPTY pkg/handlers (cached)
EMPTY pkg/logger
PASS pkg/services.TestSQLTranslation (0.01s)
PASS pkg/services.TestResourceFindByConsumerID (0.00s)
PASS pkg/services
EMPTY pkg/util
EMPTY cmd/maestro
EMPTY cmd/maestro/clone
PASS cmd/maestro/environments.TestLoadServices (0.37s)
PASS cmd/maestro/environments
EMPTY cmd/maestro/migrate
EMPTY cmd/maestro/servecmd
EMPTY cmd/maestro/server
EMPTY cmd/maestro/server/logging

DONE 10 tests in 2.590s
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
                ./test/integration
I1122 23:15:03.101294  221216 framework.go:72] Initializing testing environment
I1122 23:15:03.356261  221216 framework.go:152] Using Mock OCM Authz Client
I1122 23:15:03.356271  221216 framework.go:177] Disabling Sentry error reporting
I1122 23:15:03.360450  221216 api_server.go:145] Serving without TLS at localhost:8000
I1122 23:15:03.360461  221216 healthcheck_server.go:56] Serving HealthCheck without TLS at localhost:8083
PASS test/integration.TestConsumerPost (0.26s)
PASS test/integration.TestControllerRacing (1.42s)
PASS test/integration.TestResourceGet (0.40s)
PASS test/integration.TestResourcePost (0.25s)
PASS test/integration.TestResourcePatch (0.25s)
PASS test/integration.TestResourcePaging (0.28s)
PASS test/integration.TestResourceListSearch (0.30s)
PASS test/integration.TestUpdateResourceWithRacingRequests (0.30s)
PASS test/integration.TestUpdateResourceWithRacingRequests_WithoutLock (0.28s)
I1122 23:15:07.108731  221216 api_server.go:151] Web server terminated
PASS test/integration

DONE 9 tests in 4.800s
```

**Comment by markturansky** [comment]
squashed, rebased, and test runs posted.

**Review by clyang82 — APPROVED** [review]

**Pull request merged**